### DISCUSSION TITLE
How can I enhance the flexibility of `copilot_node_command`?

### DISCUSSION THREAD

**Discussion by sontungexpt** [zbirenbaum/copilot.lua#234] [Q&A] [answered]
Since I'm lazy loading nvm for my kitty terminal I want to include the node path directly by creating this function and calling it in `copilot_node_command`
![image](https://github.com/zbirenbaum/copilot.lua/assets/92097639/5ce9da32-4c03-4c97-9b6b-8240effb6278)

```lua
local function get_node_path()
	local node_dir = vim.fn.expand("$HOME") .. "/.nvm/versions/node"

	if vim.fn.isdirectory(node_dir) == 0 then return "node" end

	-- get the latest version of node
	local node_version = vim.fn.system("ls -v " .. node_dir .. " | tail -n 1")

	if vim.fn.empty(node_version) == 1 then return "node" end
	return node_dir .. "/" .. node_version .. "/bin/node"
end
```
But I get this error
![image](https://github.com/zbirenbaum/copilot.lua/assets/92097639/307f77f0-9190-4ab5-abec-2d1eb37bfc0d)

If I add it in this manner, everything works fine, but it lacks flexibility
![image](https://github.com/zbirenbaum/copilot.lua/assets/92097639/7049e11b-abf4-4a64-bbfc-14253c97fd95)

**Comment by MunifTanjim** [comment, marked as answer]
Are you sure `local node_version = vim.fn.system("ls -v " .. node_dir .. " | tail -n 1")` is what you expect it to be?

I think your `node_version` variable has a line break at the end of it. Try doing:

```lua
local node_version = vim.split(vim.fn.system("ls -v " .. node_dir .. " | tail -n 1"), "\n")[1]
```
